freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/dns/model/AddDnsCnameRecordRequest.java

```diff
@@ -0,0 +1,85 @@
+package com.sequenceiq.freeipa.api.v1.dns.model;
+
+import static com.sequenceiq.freeipa.api.v1.dns.model.DnsRecordRegexpPatterns.DNS_CNAME_MSG;
+import static com.sequenceiq.freeipa.api.v1.dns.model.DnsRecordRegexpPatterns.DNS_CNAME_PATTERN;
+import static com.sequenceiq.freeipa.api.v1.dns.model.DnsRecordRegexpPatterns.DNS_ZONE_MSG;
+import static com.sequenceiq.freeipa.api.v1.dns.model.DnsRecordRegexpPatterns.DNS_ZONE_PATTERN;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.sequenceiq.freeipa.api.v1.dns.doc.DnsModelDescription;
+import com.sequenceiq.service.api.doc.ModelDescriptions;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel("AddDnsCnameRecordV1Request")
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class AddDnsCnameRecordRequest {
+
+    @NotNull
+    @ApiModelProperty(value = ModelDescriptions.ENVIRONMENT_CRN, required = true)
+    private String environmentCrn;
+
+    @NotEmpty
+    @ApiModelProperty(value = DnsModelDescription.CNAME, required = true)
+    @Pattern(regexp = DNS_CNAME_PATTERN, message = DNS_CNAME_MSG)
+    private String cname;
+
+    @ApiModelProperty(DnsModelDescription.DNS_ZONE)
+    @Pattern(regexp = DNS_ZONE_PATTERN, message = DNS_ZONE_MSG)
+    private String dnsZone;
+
+    @NotEmpty
+    @ApiModelProperty(value = DnsModelDescription.CNAME_TARGET_FQDN, required = true)
+    @Pattern(regexp = "^[a-zA-Z0-9]+[a-zA-Z0-9-\\.]*[a-zA-Z0-9]+$",
```

Comment:
I think you might want the following regex: "^[a-zA-Z0-9]+[a-zA-Z0-9.-]*[a-zA-Z0-9.]+$"

1. The dash after the 9 is invalid. Put it at the front/end of the bracketed section so that it doesn't form a range.

2. Assuming the "-" is removed, this regex allows "\\" in FQDNs:
```
echo www\\google.com | egrep "^[a-zA-Z0-9]+[a-zA-Z0-9\\.]*[a-zA-Z0-9]+$"
```

3. nit: This regex technically doesn't allow a trailing "." and that technically is required to be a part of an FQDN (but it is very rare that that is actually used).

Reply:
1. I'm writing unit tests, and it seems valid

2. `\\` is necessary. first `\` is for escaping the dot, as by default dot matches anything, the second one is necessary for java to escape `\`

3. For now I will leave it this way, we can weaken the regexp later if necessary

Reply:
I have added unit tests to validate the regex. Could you check?